packages/block-editor/src/store/effects.js

```diff
@@ -219,6 +271,30 @@ export default {
 			)
 		);
 	},
+	MOVE_BLOCKS_DOWN: ( action, { getState } ) => {
```

Comment:
Can we move away from effects and instead implement these action creator as "generators"?  see `moveBlocksToPosition` as an example of action generator.

Reply:
Not sure I follow. Maybe I lack context or it's a terminology thing 🙂 In this PR we just reused actions that already exist. Are you suggesting a major refactoring in the way these actions are created?

Reply:
We're slowly moving away from using "effects" (see effects.js file and the https://github.com/aduth/refx middleware) and instead we're trying to replace them with "generator" actions (consistency with resolvers).

These are just two technologies that allow the same thing: writing complex actions with side effects. 
So ideally, new code shouldn't add new effects and use generators instead.

Reply:
Good to know. Is this documented anywhere? How new contributors are supposed to be informed about this? 🙂 If it's not documented, I'd say it's a barrier to new contributors...

Reply:
You're right, it should be. We were hoping we could get rid of these files which remove the confusion but the refactor efforts staled.